.gitlab-ci.yml

```diff
@@ -4312,6 +4312,36 @@ revert_google_container_registry_latest_7:
     - inv -e docker.publish gcr.io/datadoghq/dogstatsd:${NEW_LATEST_RELEASE_7} gcr.io/datadoghq/dogstatsd:latest
     - inv -e docker.publish gcr.io/datadoghq/dogstatsd:${NEW_LATEST_RELEASE_7} gcr.io/datadoghq/dogstatsd:7
 
+# Once a day, before the nightly build, cleans up the artifacts used during kitchen tests which might have been left over
+# This can happen when a kitchen test fails and is never retried, since that pipeline's cleanup job won't run
+periodic_kitchen_cleanup_s3:
+  stage: maintenance_jobs
+  tags: [ "runner:main", "size:large" ]
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-builders/gitlab_agent_deploy:$DATADOG_AGENT_BUILDERS
+  rules:
+    - <<: *if_deploy_on_nightly
+  script:
+    - aws s3 rm --recursive s3://$DEB_TESTING_S3_BUCKET/pool
+    - aws s3 rm --recursive s3://$DEB_TESTING_S3_BUCKET/dists
+    - aws s3 rm --recursive s3://$RPM_TESTING_S3_BUCKET/testing/
+    - aws s3 rm --recursive s3://$RPM_TESTING_S3_BUCKET/suse/testing/
+    - aws s3 rm --recursive s3://$WIN_S3_BUCKET/pipelines/A6/
+    - aws s3 rm --recursive s3://$WIN_S3_BUCKET/pipelines/A7/
```

Comment:
Should `inv pipeline.trigger` be changed to match this? The invoke task defaults to `nightly`, which means that someone using it to test something would trigger a cleanup (in the middle of the day, which would probably cause failures for other pipelines.

An alternative would be to have a separate `TESTING_CLEANUP` variable, that only the periodic nightly trigger would add, which would trigger this job.

Reply:
Good point. The latter makes more sense to me.